EIPS/eip-5018.md

```diff
@@ -0,0 +1,155 @@
+```
+
+#### readChunk
+
+Returns the chunk data of the file `name` and the existence of the chunk.
+
+```
+function readChunk(bytes memory name, uint256 chunkId) external view returns (bytes memory chunkData, bool exist);
+```
+
+#### chunkSize
+
+Returns the size of a chunk of the file `name` and the existence of the chunk.
+
+```
+function chunkSize(bytes memory name, uint256 chunkId) external view returns (uint256 chunkSize, bool exist);
+```
+
+#### removeChunk
+
+Removes a chunk of the file `name` and returns `false` if such chunk does not exist. The method should be called by an account with write permission.
+
+```
+function removeChunk(bytes memory name, uint256 chunkId) external returns (bool exist);
+```
+
+#### truncate
+
+Removes the chunks of the file `name` in the directory from the given `chunkId` and returns the number of chunks removed by an account with write permission.  When `chunkId = 0`, the method is essentially the same as `remove()`.
+
+```
+function truncate(bytes memory name, uint256 chunkId) external returns (uint256 numOfChunksRemoved);
+```
+
+#### getChunkHash
+
+Returns the hash value of the chunk data.
+```
+function getChunkHash(bytes memory name, uint256 chunkId) external view returns (bytes32);
+```
+
+## Rationale
+
+One issue of uploading the web contents to the blockchain is that the web contents may be too large to fit into a single transaction.  As a result, the standard provides a chunk-based operations so that uploading a content can be splitted into several transactions.  Meanwhile, the read operation can be done in a single transaction, i.e., with a single Web3 URL defined in EIP-4804.
+
+### Interactions Between Unchunked/Chunked Functions
+
+`read` method should return the concatenated chunked data written by `writeChunk` method.  The following gives some examples of the interactions:
+
+- `read("hello.txt")` => "" (file is empty)
+- `writeChunk("hello.txt", 0, "abc")`  will succeed
+- `read("hello.txt")` => "abc"
+- `writeChunk("hello.txt", 1, "efg")`  will succeed
+- `read("hello.txt")` => "abcefg"
+- `writeChunk("hello.txt", 0, "aaa")`  will succeed (replace chunk 0's data)
+- `read("hello.txt")` => "aaaefg"
+- `writeChunk("hello.txt", 3, "hij")`  will fail because the operation is not replacement or append.
+
+With `writeChunk` method, we allow writing a file with external data that exceeds the current calldata limit (e.g., 1.8MB now, and 1.4MB post-EIP-4488), and it is able to read the whole file in a single `read` method (which is friendly for large web objects such as HTML/SVG/PNG/JPG, etc).
+
+For `write` method, calling a `write` method will replace all data chunks of the file with `write` method data, and one implementation can be:
+1. `writeChunk(filename, chunkId=0, data_from_write)` to chunk 0 with the same `write` method data; and
+2. `truncate(filename, chunkId=1)`, which will remove the rest chunks.
+
```

Comment:
Missing required backward compatibility section.

Reply:
Thanks for the comment.  The backward compatibility section was added.